### PR TITLE
Add test for `flexible_space_bar.0.dart`

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds a test for
- `examples/api/test/material/flexible_space_bar/flexible_space_bar.0_test.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by bleroux** [comment]
I remember having a look at this example. The issue with this kind of examples is that they are not usable on DartPad for most people (no drag).
Can you update the original example [following these instructions](https://github.com/flutter/flutter/pull/129377#pullrequestreview-1494105152)?

And then update the test, it would probably be great to run it on all platforms.

**Comment by ValentinVignal** [comment]
@bleroux sure ! I made the change in [revamp: Allow all pointers to scroll](https://github.com/flutter/flutter/pull/157107/commits/ec602ede0ebaade3db3a0f65b5a47b3b20c226c2)

File: examples/api/test/material/flexible_space_bar/flexible_space_bar.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    final Size sizeBeforeScroll = tester.getSize(appBarContainer);
+    final Offset target = tester.getCenter(find.byType(ListTile).first);
+    final TestGesture gesture = await tester.startGesture(target);
+    await gesture.moveBy(const Offset(0.0, 100.0));
+    await tester.pump(const Duration(milliseconds: 10));
+    await gesture.up();
+    final Size sizeAfterScroll = tester.getSize(appBarContainer);
+
+    expect(sizeBeforeScroll.height, lessThan(sizeAfterScroll.height));
+  });
```
Comment (bleroux) [inline examples/api/test/material/flexible_space_bar/flexible_space_bar.0_test.dart:40]: Can you try to activate this test on all platforms?
(Maybe with a comment mentioning this is to test that ScrollBehavior.dragDevices is correctly set).

File: examples/api/test/material/flexible_space_bar/flexible_space_bar.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // The app being tested loads images via HTTP which the test
+  // framework defeats by default.
+  setUpAll(() {
+    HttpOverrides.global = null;
+  });
+
+  testWidgets('The app bar should stretch when over-scrolled', (WidgetTester tester) async {
```
Comment (bleroux) [inline examples/api/test/material/flexible_space_bar/flexible_space_bar.0_test.dart:18]:
```suggestion
  testWidgets('The app bar stretches when over-scrolled', (WidgetTester tester) async {
```

I was used to add 'should' on test descriptions but it looks better without it. :smile:

File: examples/api/test/material/flexible_space_bar/flexible_space_bar.0_test.dart (hunk 1 above)
Comment (ValentinVignal) [reply examples/api/test/material/flexible_space_bar/flexible_space_bar.0_test.dart]: Sure! Is [test: Activate all platforms](https://github.com/flutter/flutter/pull/157107/commits/af9653e54d5729f96fa14910d47f39a7ebb96511) what you expected ?

**Review by bleroux — APPROVED** [review]
LGTM!

**Pull request merged**